test/expectations/expectations.go

```diff
@@ -78,18 +77,24 @@ func ExpectApplied(ctx context.Context, c client.Client, objects ...client.Objec
 			o.SetResourceVersion(current.GetResourceVersion())
 			Expect(c.Update(ctx, o)).To(Succeed())
 		}
-		// Update status
-		statuscopy.SetResourceVersion(o.GetResourceVersion())
-		Expect(c.Status().Update(ctx, statuscopy)).To(Or(Succeed(), MatchError(ContainSubstring("not found")))) // Some objects do not have a status
 
 		// Re-get the object to grab the updated spec and status
 		Expect(c.Get(ctx, client.ObjectKeyFromObject(o), o)).To(Succeed())
 	}
 }
 
+func ExpectStatusUpdated(ctx context.Context, c client.Client, objects ...client.Object) {
+	GinkgoHelper()
+	for _, o := range objects {
+		Expect(c.Status().Update(ctx, o.DeepCopyObject().(client.Object))).To(Succeed())
+		Expect(c.Get(ctx, client.ObjectKeyFromObject(o), o)).To(Succeed())
```

Comment:
Wouldn't we want to get the object first then update it?

Reply:
If we got the object first, we would override our desired object to be written, and the update would be a noop. 

We read afterwards to re-hydrate any mutating admission changes made to the object on the update, so we don't have a slightly-out-of-sync object. We also do this on create and delete, as the controller-runtime client only updates the resource version, and isn't aware of any of the server-side mutated fields.

Reply:
Ah I wasn't aware of the re-hydration. That's smart. On getting the object first, @ellistarn and I synced offline. Looks like the trade-off here is using merge patches has the inability to nil out non-pointer strings, whereas update suffers from optimistic locking. If we can make sure we're doing a Get call before the Update, like we currently are, I think that's the better path.